pyscitt/pyscitt/verify.py

```diff
@@ -167,30 +166,109 @@ class DynamicTrustStore(TrustStore):
         keys = {key["kid"].encode(): key for key in jwks}
         key = keys[parsed.phdr[KID]]
         pem_key = jwk.JWK.from_json(json.dumps(key)).export_to_pem()
         key = load_pem_public_key(pem_key, default_backend())
-        return key
+        return key  # type: ignore
+
+
+class DynamicTrustStoreClient:
+    """
+    A client for retrieving public keys from a transparency service identified by an issuer.
+    """
+
+    clients: Dict[str, httpx.Client] = {}
+
+    def __init__(self, forced_httpclient: Optional[httpx.Client] = None):
+        self.retries = 5
+        # forced client is used in testing
+        self._forced_httpclient = forced_httpclient
+
+    def _client(self, cadata: Optional[str] = None) -> httpx.Client:
+        if self._forced_httpclient is not None:
+            return self._forced_httpclient
+
+        if cadata:
+            ssl_context = ssl.create_default_context()
+            ssl_context.load_verify_locations(cadata=cadata)
+            transport = httpx.HTTPTransport(verify=ssl_context, retries=self.retries)
+        else:
+            transport = httpx.HTTPTransport(retries=self.retries)
+        return httpx.Client(transport=transport)
+
+    def _client_for_issuer(self, issuer: str) -> httpx.Client:
+        """
+        Lightweight caching of clients based on issuer.
+        """
+        if issuer not in self.clients:
+            if self.is_confidential_ledger_issuer(issuer):
+                pem = self.get_confidential_ledger_tls_pem(issuer)
+                self.clients[issuer] = self._client(cadata=pem)
+            else:
+                self.clients[issuer] = self._client()
+        return self.clients[issuer]
+
+    def get_jwks(self, issuer: str) -> httpx.Response:
+        """
+        Retrieve the JWKS from the issuer.
+        """
+        transparency_configuration = self.get_configuration(issuer)
+        transparency_configuration.raise_for_status()
+        config_response = cbor2.loads(transparency_configuration.content)
+        jwks_uri = config_response["jwks_uri"]
+        return self._client_for_issuer(issuer).get(
+            jwks_uri, headers={"Accept": "application/json"}
+        )
+
+    def get_configuration(self, issuer: str):
+        """
+        Retrieve the transparency configuration from the issuer.
+        """
+        if not issuer:
+            raise ValueError("Issuer cannot be empty")
+
+        return self._client_for_issuer(issuer).get(
+            "https://" + issuer + "/.well-known/transparency-configuration",
+            headers={"Accept": "application/cbor"},
+        )
+
+    def is_confidential_ledger_issuer(self, issuer: str) -> bool:
+        return issuer.endswith(".confidential-ledger.azure.com")
+
+    def get_confidential_ledger_tls_pem(self, issuer: str):
+        """
+        Retrieve the TLS certificate for a Confidential Ledger issuer.
+        """
+        if not self.is_confidential_ledger_issuer(issuer):
```

Comment:
nit: this check seems redundant since the caller of this method is already calling the same method.